astropy/modeling/mappings.py

```diff
@@ -182,3 +183,117 @@ def inverse(self):
         """
 
         return self
+
+
+class UnitsMapping(Model):
+    """
+    Coerce input units (or dimensionless input) into new units without
+    conversion.  Used by Model.coerce_units to support units on otherwise
+    dimensionless models such as Polynomial1D.
+
+    Parameters
+    ----------
+    mapping : tuple
```

Comment:
This does not really sound like a "true" `mapping`. Maybe just have two parameters like `input_units`, `output_units` (or so).

Reply:
I was imagining this model as "mapping" one unit to another -- at the first input position, meters --> dimensionless, at the second input position, seconds --> dimensionless, etc.  In that example, the mapping parameter would look like this:

```python
mapping=((units.m, units.dimensionless_unscaled), (units.s, units.dimensionless_unscaled))
```

Is this your suggestion for separate parameters?

```python
input_units=(units.m, unit.s)
output_units=(units.dimensionless_unscaled, units.dimensionless_unscaled)
```

I have a slight preference for the `mapping` version, only because it's easier on my brain to see the input and output units for the same parameter side-by-side.

Reply:
I think it is fine as is as long as it is clear from the documentation how things work.

Reply:
But yes, your example is what I was thinking about.

Reply:
Also, please consider, an option to get just the "numbers" or values, i.e., 
```python
mapping=((units.m, None), (units.s, None))
```
if one would like to get rid of `Quantities` altogether.

Reply:
Thanks for the reminder, I forgot we talked about that.

Reply:
I made a commit, `None` should now tear of the units entirely.